code/Model/SiteTree.php

```diff
@@ -1740,7 +1740,7 @@ public function validURLSegment()
         // Check known urlsegment blacklists
         if (self::config()->get('nested_urls') && $this->ParentID) {
             // Guard against url segments for sub-pages
-            $parent = $this->Parent();
+            $parent = $this->getParent();
             if ($controller = ModelAsController::controller_for($parent)) {
```

Comment:
If there's no parent, we're passing in null - but the typehinting for `ModelAsController::controller_for` requires a `SiteTree` object.

There are probably other situations like this that need to be accounted for which is probably why tests are failing.

Reply:
Yes, I completely rewrote my work now as there were tons of errors.

Here is what I did in your instance:
https://github.com/silverstripe/silverstripe-cms/pull/2709/commits/c849252512ab83c60f068d197c02896a684365a7#diff-08b08699fad031e3fb13c3637d1a2b0a71a0b82f2bd69606b2244b12f89b6129L1741-L1747